### ISSUE TITLE
springboot-zip example not working properly with sam local

### ISSUE THREAD

**Issue by jschwartz73** [awslabs/aws-lambda-web-adapter#420]
Hello,

I've been trying to run the springboot-zip example for the past few days.  I had to make a minor change to the template to get it to work with my m2 mac.  Everything appears to be running, however it doesn't appear that the api gateway routes are being mapped from `/v1/{proxy+}` to the proper route in the springboot app (in my case `/pets`).  All `/v1` routes return a 404.  

Additionally there are numerous exceptions in the springboot app logs.  Sometimes a request will not generate an exception, sometimes the request returns one kind of exception and another time it will generate a different exception.  My hunch is the springboot exceptions have more to do with the lack of mapping mentioned above.  

Regardless, all requests return a 404

My environment:

```
Maven home: /Users/jeff/Dropbox/third/apache-maven-3.9.6
Java version: 17.0.1, vendor: Amazon.com Inc., runtime: /Library/Java/JavaVirtualMachines/amazon-corretto-17.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "14.4", arch: "aarch64", family: "mac"

aws-cli/2.15.35 Python/3.11.8 Darwin/23.4.0 exe/x86_64 prompt/off

SAM CLI, version 1.113.0

Docker version 25.0.3, build 4debf41
Docker desktop: v4.28.0
``` 

To get the example to run on the M2 mac, I had to add the following to the template.yaml
```
Architectures:
        - arm64
``` 

But i also made some minor changes to the springboot app to enhance the logs for troubleshooting.

I also captured all of the output in the following files:

1. build.out.txt: output of `sam build --debug`
2. start-api.out.txt: output of `sam local start-api --debug --log-file app-runtime.log`
3. app-runtime.log: output of springboot app (on trace log level)

I executed the following from postman about a half dozen times:

`GET http://127.0.0.1:3000/v1/pets`

All of the above changes and files can be found on this branch

https://github.com/awslabs/aws-lambda-web-adapter/compare/main...SchwarTech:aws-lambda-web-adapter:main

Any help would be greatly appreciated.

Thanks in advance.

**Comment by jschwartz73** [comment]
Additional information.  The two exceptions that appear in the springboot's app log (included in my branch as app-runtime.log)

1. org.apache.tomcat.jni.LibraryNotFoundError: Can't load library: /var/task/bin/libtcnative-1.so
2. thread 'main' panicked at src/main.rs:25:25:
lambda runtime failed: Error { kind: SendRequest, source: Some(hyper::Error(IncompleteMessage)) }

**Comment by bnusunny** [comment]
I think this is the key: 

`org.apache.tomcat.jni.LibraryNotFoundError: Can't load library: /var/task/bin/libtcnative-1.so`

I don't think the example has this native library file.

**Comment by jschwartz73** [comment]
Thanks for the quick response.

i installed the tomcat-native library via brew and re-built the springboot-zip example.

Now about 80% of the postman requests end up with an `internal server error` and the same error as above:
* http: panic serving 127.0.0.1:56700: runtime error: invalid memory address or nil pointer dereference
goroutine

Updated [app-runtime.log](https://github.com/awslabs/aws-lambda-web-adapter/files/14882639/app-runtime.log) file attached.

Any feedback would be greatly appreciated.

Thanks in advance.

**Comment by bnusunny** [comment]
I think there is one issue: the SpringBoot app is listening to port `8080`. This conflicts with Lambda Runtime Interface Emulator. You need to change the `run.sh` and configurate an environment variable `PORT=8000`. 

`exec java -cp "./:lib/*" "com.amazonaws.demo.petstore.Application" --server.port=$PORT`

**Comment by bnusunny** [comment]
BTW, for local development, you could run the SpringBoot app directly without sam local. After all, LWA aims to enable you using the familiar tools to local development.

**Comment by jschwartz73** [comment]
I tried using port 8081 for the spring boot app and got the same results.
I believe there is a problem with SAM local with x86_64 images on a Mac.

Since I'm actually not using spring boot, I implemented an example using
Javalin 6 and ARM64 (PR submitted).  It works great in sam local and aws
lambda.  I was also able to get the Javalin app to run in aws lambda using
x86_64, but it too would not run in sam local.

I do not plan on using sam local for development, it was more for
learning the project and seeing how it works w/Java w/o deploying to aws.

Thanks in advance for everyone's help
